runtime/doc/lsp.txt

```diff
@@ -2340,7 +2342,7 @@ start({cmd}, {dispatchers}, {extra_spawn_params})        *vim.lsp.rpc.start()*
                                 See |vim.system()|
 
     Return: ~
-        (`vim.lsp.rpc.PublicClient?`) Client RPC object, with these methods:
+        (`vim.lsp.rpc.PublicClient`) Client RPC object, with these methods:
```

Comment:
Pardon my ignore here, but does this change in the return type mean that in order for a plugin to check if creating an LSP client succeeds, they can no longer rely on checking if this returns `nil` or not and instead have to wrap the call with a `pcatch`?

Reply:
Yes. Need to do a proper code search, but as far as I’m aware no plugins are using this API directly except nvim-lspconfig, which I will update myself when this is merged.

Alternatively we could use multi return values, `client|nil, nil|string` where the 2nd argument is the error if the client failed to start. That’s still a breaking change, but slightly less breaking.

Reply:
Got it, thanks for the early research. I think the current implementation is good for now, but given that `vim.lsp.start` wasn't added in v0.10, the change at least deserves an entry in `news.txt`.